File: validParentheses.java
import java.util.Vector;
public class Solution {
    public boolean isValid(String s) {
        // Start typing your Java solution below
        // DO NOT write main() function
        Vector<Character> v = new Vector<Character>();
        for (int i=0; i < s.length(); i++) {
            char a = s.charAt(i);
            if (a == '(' || a == '{' || a == '[') {
                v.add(a);
            }
            else {
                if (v.size() == 0) {
                    return false;
                }
                char c = v.lastElement();
                v.remove(v.size()-1);
                if (c == '(' && a == ')') {
                    continue;
                }
                if (c == '{' && a == '}') {
                    continue;
                }
                if (c == '[' && a == ']') {
                    continue;
                }
                return false;
            }
        }
        if (v.size() == 0) {
            return true;
        }
        else {
            return false;
        }
        
    }
}

File: validParentheses.cpp
#include "leetcode.h"
class Solution {
public:
    bool isValid(string s) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        vector<char> v;
        for (int i=0; i < s.size(); i++) {
            if (s[i] == '(' || s[i] == '{' || s[i] == '[') {
                v.push_back(s[i]);
            }
            else {
                if (v.empty()) {
                    return false;
                }
                char c = v.back();
                v.pop_back();
                if (c == '(' && s[i] == ')') {
                    continue;
                }
                if (c == '{' && s[i] == '}') {
                    continue;
                }
                if (c == '[' && s[i] == ']') {
                    continue;
                }
                return false;
            }
        }
        if (v.empty()) {
            return true;
        }
        else {
            return false;
        }
    }
};
